bundles/org.openhab.binding.bosesoundtouch/src/main/java/org/openhab/binding/bosesoundtouch/internal/handler/BoseSoundTouchHandler.java

```diff
@@ -359,96 +365,113 @@ public PresetContainer getPresetContainer() {
     }
 
     @Override
-    public void onWebSocketConnect(Session session) {
+    public void onWebSocketConnect(@Nullable Session session) {
         logger.debug("{}: onWebSocketConnect('{}')", getDeviceName(), session);
         this.session = session;
         commandExecutor = new CommandExecutor(this);
         updateStatus(ThingStatus.ONLINE);
     }
 
     @Override
-    public void onWebSocketError(Throwable e) {
-        logger.debug("{}: Error during websocket communication: {}", getDeviceName(), e.getMessage(), e);
-        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-        if (commandExecutor != null) {
-            commandExecutor.postOperationMode(OperationModeType.OFFLINE);
+    public void onWebSocketError(@Nullable Throwable e) {
+        Throwable localThrowable = (e != null) ? e : new Throwable("Null Exception passed to onWebSocketError");
```

Comment:
It's usually better to create exceptions of a specific type:

```suggestion
        Throwable localThrowable = (e != null) ? e : new IllegalStateException("Null Exception passed to onWebSocketError");
```